napari/utils/_proxies.py

```diff
@@ -98,3 +144,50 @@ def create(cls, obj: Any) -> Union['PublicOnlyProxy', Any]:
 class CallablePublicOnlyProxy(PublicOnlyProxy[Callable]):
     def __call__(self, *args, **kwargs):
         return self.__wrapped__(*args, **kwargs)
+
+
+def in_main_thread_py() -> bool:
+    """
+    Check if caller is in main python thread.
+
+    Returns
+    -------
+    thread_flag : bool
+        True if we are in the main thread, False otherwise.
+    """
+    import threading
+
+    return threading.current_thread() == threading.main_thread()
+
+
+def _in_main_thread() -> bool:
+    """
+    General implementation of checking if we are in a proper thread.
+    If Qt is available and Application is created then assign :py:func:`in_qt_main_thread` to `in_main_thread`.
+    If Qt liba are not available then assign :py:func:`in_main_thread_py` to in_main_thread.
+    IF Qt libs are available but there is no Application ti wil emmit warning and return result of in_main_thread_py.
+
+    Returns
+    -------
+    thread_flag : bool
+        True if we are in the main thread, False otherwise.
+    """
+
+    global in_main_thread
```

Comment:
Why do we need the global? Is this so that we only execute this function once for performance reasons?

This function is only called when an env variable is set, so I don't think we need to worry to much about performance. So if that's the only reason to do this, I'd suggest removing the global and just defining this function as `in_main_thread`.

Reply:
Yes